apps/lockfile-explorer/src/start.ts

```diff
@@ -103,7 +105,30 @@ function startApp(debugMode: boolean): void {
 
   app.get('/api/lockfile', async (req: express.Request, res: express.Response) => {
     const pnpmLockfileText: string = await FileSystem.readFileAsync(appState.pnpmLockfileLocation);
-    const doc = yaml.load(pnpmLockfileText);
+    const doc = yaml.load(pnpmLockfileText) as Lockfile;
+    const { packages, lockfileVersion } = doc;
+
+    let shrinkwrapFileMajorVersion: number;
+    if (typeof lockfileVersion === 'string') {
+      shrinkwrapFileMajorVersion = parseInt(lockfileVersion.substring(0, lockfileVersion.indexOf('.')), 10);
```

Comment:
This code assumes the version contains `.` and that it is a valid integer.  Otherwise `parseInt()` can return `NaN` which will fail the test `if (shrinkwrapFileMajorVersion < 5 || shrinkwrapFileMajorVersion > 6)`.

Generally speaking, we don't need to worry about variables not conforming to their declared TypeScript type or expected value.  However in this case, the input file has not been validated with a JSON schema, and the version of its file format  is also unknown.  Thus some extra validation here seems worthwhile.